src/ncp/ncp_base.cpp

```diff
@@ -2409,6 +2409,8 @@ extern "C" void otNcpPlatLogv(otLogLevel aLogLevel, otLogRegion aLogRegion, cons
             charsWritten = static_cast<int>(sizeof(logString) - 1);
         }
 
+        logString[charsWritten++] = '\n';
```

Comment:
@simonlingoogle 

We have `OPENTHREAD_CONFIG_LOG_SUFFIX` config which can be used for this purpose (a suffix that to be added to end of all log messages by the logging modules/macros).
https://github.com/openthread/openthread/blob/75a382d292b2271c0a6e202324550091753467f0/src/core/config/logging.h#L291-L299

The OpenThread NCP build may not be used with `wpantund` (e.g. it may be used by an RTOS-based driver) and always including `\n` may not be desired by all host drivers, so I won't recommend adding it here.

Also if you are using NCP build with `wpantund` I would highly suggest using the config
`OPENTHREAD_CONFIG_LOG_OUTPUT` to `OPENTHREAD_CONFIG_LOG_OUTPUT_NCP_SPINEL`.

Reply:
Thanks for your suggestions. This PR is not thoughtful enough.